aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/EventLoopGroupBuilder.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import java.util.concurrent.ThreadFactory;
+import software.amazon.awssdk.utils.builder.SdkBuilder;
+
+/**
+ * Builder for a Netty {@link EventLoopGroup}. Will use the optimal {@link EventLoopGroup} for the
+ * given platform.
+ */
+public final class EventLoopGroupBuilder {
+
+    /**
+     * Create a {@link Builder} that can be used to create an immutable {@link EventLoopGroupBuilder}.
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
```

Comment:
I like the factory approach. More consistent and gives then additional functionality if they just want to configure the event loop but still want the SDK to manage it (i.e. shut it down on close). Which reminds me I need to make the event loop unmanaged when the pass it in explicitly.